numpy/_core/src/multiarray/methods.c

```diff
@@ -1281,6 +1285,21 @@ array_sort(PyArrayObject *self,
         newd->names = new_name;
         ((PyArrayObject_fields *)self)->descr = newd;
     }
+    if (descending) {
+        PyErr_SetString(PyExc_ValueError,
+            "`descending=True` is not allowed. Use `np.flip` instead");
+        return NULL;
```

Comment:
As @mhvk mentioned in the comment, it could be supported by negating the stride, but if it's Ok I would try to add it in a separate PR, as it's mostly C-level change.

Reply:
Honestly, this change is tiny.  If we don't like it on its own, we should do the full work immdiately.

It must have been mentioned before many times, but flipping is also *not* correct for stable argsort!

Reply:
Thank you for the guidance! Is it Ok to do the `flip` in the Python level (using existing `np.flip`) pass to C's `array_sort`, and flip back at Python level? Or should I redo `flip` in `array_sort`?

Reply:
Thinking about this a bit more, I agree with @seberg that if we add a new argument, we should at least support it, even if the support is not yet optimal. It may make sense to do a python implementation (with some tests!) if that's easier. With a big TODO that the proper answer would seem to be to just allow sorting using either `<` or `>` (`Tag::less` in `quicksort.cpp`, simply passing on `ascending/descending` to highway, quite possibly equally easy for other methods; only the one from the descriptor would need special-casing). 

Note that `argsort` is a bit tricky: equivalent to `n - array[::-1].argsort()[::-1]` (negating the array and adding the number of points `n` along an axis could be done in-place).  Check:
```
a = np.array([0, 1, 2, 2, -1, -1])
a.argsort()
# array([4, 5, 0, 1, 2, 3])

5-a[::-1].argsort()[::-1]
# array([2, 3, 1, 0, 4, 5])
```

Reply:
Yeah, sorry, I hadn't realized the compat library already has these hacks correctly in place...  even without extending the core sorting, it may be nice to do the flipping in C (although I am not sure we have a helper to make it easy).  Otherwise the method and function diverge, which isn't great.

Reply:
`np.flip` is pure python, and uses slices `::-1` to reverse the order. Which is implemented as https://github.com/numpy/numpy/blob/0ebbdb9f24f96fc97b8df1ecb632e23a86b65625/numpy/_core/src/multiarray/mapping.c#L872-L874

Reply:
FYI `descending` will be addressed in a separate PR.